include/boost/pool/pool.hpp

```diff
@@ -360,7 +360,7 @@ class pool: protected simple_segregated_storage < typename UserAllocator::size_t
     size_type max_chunks() const
     { //! Calculated maximum number of memory chunks that can be allocated in a single call by this Pool.
       size_type POD_size = integer::static_lcm<sizeof(size_type), sizeof(void *)>::value + sizeof(size_type);
-      return (std::numeric_limits<size_type>::max() - POD_size) / alloc_size();
+      return ((std::numeric_limits<size_type>::max)() - POD_size) / alloc_size();
```

Comment:
Is this a better workaround than the use of `BOOST_PREVENT_MACRO_SUBSTITUTION` elsewhere in this file?

Reply:
For example on line 412

Reply:
I used the same as line 351

https://github.com/boostorg/pool/blob/e0db4278cafecd820e3575877f17afcbd441110c/include/boost/pool/pool.hpp#L351

Reply:
I think in use `BOOST_PREVENT_MACRO_SUBSTITUTION` but it extends the line length and reduce readability.

Reply:
I disagree.
The bare parentheses fix the problem, and they're small.
But they give no clue as to why they are there - and that's my objection.

In the future, someone will come along and remove the "redundant" parens.
Hopefully, when that happens, this whole "windows header" thing will be a bad memory.

Reply:
Would be great if this could be merged. As when compiling a project with boost 1.82 or 1.83 I run into the following warning many times: `C4003: not enough arguments for function-like macro invocation 'max'` and this seems to be the culprit.

I also found an issue in another boost lib that addresses the same problem this way.
https://github.com/boostorg/beast/pull/1981/commits/24a1196e4c36ac34b0072876075f3dba9bec1d35
Would it help to add a comment to explicitly state why those parentheses are there?